### PR TITLE
MAGN-6615: Kill ProtoInterface

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
### Purpose

This PR is a follow-up of [LibG changes to remove ProtoInterface](https://git.autodesk.com/Dynamo/LibG/pull/251). The core interfaces required by Dynamo zero touch library developers and application integrators are moved from ProtoInterface to DynamoServices project.

DynamoServices.dll will be published as [DynamoVisualProgramming.DynamoServices NuGet package](https://www.nuget.org/packages/DynamoVisualProgramming.DynamoServices).

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

- [x] @ikeough 
- [ ] @Randy-Ma 
- [ ] @aparajit-pratap 

### FYIs
- [ ] @mjkkirschner 
- [ ] @ke-yu 
- [ ] @Benglin 
- [ ] @pboyer

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@sharadkjaiswal just a thought-  do the changes here need to go into the https://github.com/DynamoDS/Dynamo/wiki/API-Changes since they will effect zero touch node authors?

File: src/NodeServices/Interfaces.cs [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        /// <summary>
+        /// Gets the context provider for interpretation of data in designscript
+        /// world.
+        /// </summary>
+        IContextDataProvider ContextProvider { get; }
+    }
+
+    /// <summary>
+    /// Represents a connector to external data source to provide context 
+    /// speicific data. This interface provide import/export feature for any
```
Comment (ikeough) [inline src/NodeServices/Interfaces.cs]: "specific"

File: src/NodeServices/Interfaces.cs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    /// </summary>
+    public interface IContextDataProvider
+    {
+        /// <summary>
+        /// Gets the name of this data provider to identify it uniquely.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Imports specific the context data using the given identifier string.
```
Comment (ikeough) [inline src/NodeServices/Interfaces.cs]: Remove "specific."

File: src/NodeServices/Interfaces.cs [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        /// </summary>
+        /// <param name="connectionParameters">Input dictionary of connection parameters 
+        /// to connect to the data source to import the data. Each context data in 
+        /// the list contains pair of connection parameter name and value</param>
+        /// <returns></returns>
+        IContextData[] ImportData(Dictionary<string, Object> connectionParameters);
+
+        /// <summary>
+        /// Exports data to specified file. This context provider determines the
```
Comment (ikeough) [inline src/NodeServices/Interfaces.cs]: "to the specified file."

**Pull request merged**